org.lflang/src/org/lflang/generator/ReactionInstanceGraph.java

```diff
@@ -254,4 +277,43 @@ private void assignLevels() {
             removeNode(origin);
         }
     }
+    
+    /**
+     * Assign a level to a reaction runtime instance.
+     * 
+     * @param runtime The reaction runtime instance.
+     * @param level The level to assign.
+     */
+    private void assignLevel(Runtime runtime, Integer level) {
+        runtime.level = level;
+        adjustMaxNumOfReactionPerLevel(level, 1);
+    }
+    
+    /**
+     * Update the level of the reaction <code>runtime<code> instance
```

Comment:
What is the rationale behind this logic?

Reply:
The level assignment algorithm could reassign a new level to a reaction that was previously assigned a smaller level (based on a newly found `origin`). If that happens, we need to adjust the values in `adjustMaxNumOfReactionPerLevel` accordingly.